src/Utils/AuthorizeFor.tsx

```diff
@@ -26,3 +35,15 @@ export const Anyone = () => true;
 export default function (userTypes: UserRole[]) {
   return (userType: UserRole) => userTypes.includes(userType);
 }
+
+export const AuthorizeUserRoute: React.FC<AuthorizeUserRouteProps> = ({
+  userTypes,
+  children,
+}) => {
+  const authUser = useAuthUser();
+  if (userTypes.includes(authUser.user_type)) {
+    return <>{children}</>;
+  } else {
+    return <Error404 />;
```

Comment:
`In future,` maybe we can consider having a 403 Forbidden page, to better represent the error in case they do not have permissions but are shared with a URL. In case someone thinks they should have access but does not, they will get to a 404 page which indicates that the issue is with the page itself, it does not exist. In this case, if the error page was 403, they would be able to communicate to the proper channel that they do not have permissions for the page, instead of the page simply not existing.

Reply:
It would be better if we create a separate PR for this
as for a 403 page, we also need an icon etc

![image](https://github.com/coronasafe/care_fe/assets/98876115/ae030800-d7db-4ccb-950f-9c04a1648834)
something like this

Reply:
@AshrafMd-1 Yep, let's do it separately in a different issue and PR.